src/view/items/shared/EventDirective.js

```diff
@@ -120,10 +121,14 @@ export default class EventDirective {
 			const result = this.fn.apply( ractive, values ).pop();
 
 			// Auto prevent and stop if return is explicitly false
-			let original;
-			if ( result === false && event && ( original = event.original ) ) {
-				original.preventDefault && original.preventDefault();
-				original.stopPropagation && original.stopPropagation();
+			if ( result === false ) {
+				const original = event ? event.original : undefined;
+				if ( original ) {
+					original.preventDefault && original.preventDefault();
+					original.stopPropagation && original.stopPropagation();
+				} else {
+					warnIfDebug( 'Handler returns false, but no event object is available' );
```

Comment:
I'd suggest including the event directive name as well, perhaps as something like:
```js
warnOnceIfDebug( `${this.template.n} handler returned false, but there is no event available to cancel` );
```